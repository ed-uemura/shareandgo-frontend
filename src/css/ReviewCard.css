:root {
	--SGGreen: #01d95e;
	--SGDarkGreen: #1e874b;
	--SGBlack: #171921;
	--SGAlpha: rgba(0, 0, 0, 0);
	--SGFont: 'Open Sans', sans-serif;
	--SGTitleSize: 1.5em;
	--SGTextAssetSize: 1em;
	--SGTextSize: 1em;
	--SGTextDetails: 0.75em;
}

.passenger-picture {
	display: flex;
	background-color: var(--SGGreen);
	width: 42px;
	height: 42px;
	border-radius: 100%;
	overflow: hidden;
	justify-content: center;
	margin: 0;
	left: 0;
}

.passenger-picture img {
	height: 100%;
}
