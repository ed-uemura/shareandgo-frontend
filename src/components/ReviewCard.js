import React from 'react';
import '../css/ReviewCard.css';
import RatingStars from './RatingStars';

const ReviewCard = ({ image, name, rating, comment }) => {
	return (
		<div>
			<div className='d-flex align-items-center mb-2'>
				<figure className='passenger-picture'>
					<img src={image} alt={name} />
				</figure>

				<div className='ms-2'>
					<p>{name}</p>
					<RatingStars rating={rating} />
				</div>
			</div>

			<div className='col-12'>
				<p>{comment}</p>
			</div>
		</div>
	);
};

export default ReviewCard;
