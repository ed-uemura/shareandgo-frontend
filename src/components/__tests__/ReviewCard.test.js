import {render, screen, cleanup} from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReviewCard from '../ReviewCard';
import DriverImage from '../../images/profile-picture-mockup.jpg';


afterEach(() =>{
    cleanup();
});

test('matches snapshot', () =>{
    const comment = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus convallis est sit.'
    const name ='Cara Moon';
    const image = DriverImage;
    const rating  = '3';
    const tree = renderer.create(<ReviewCard rating={rating} image = {DriverImage} name = {name} comment = {comment}/>).toJSON();
    expect(tree).toMatchSnapshot();
});

