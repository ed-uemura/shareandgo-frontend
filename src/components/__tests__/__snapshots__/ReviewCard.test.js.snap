// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    className="d-flex align-items-center mb-2"
  >
    <figure
      className="passenger-picture"
    >
      <img
        alt="Cara Moon"
        src="profile-picture-mockup.jpg"
      />
    </figure>
    <div
      className="ms-2"
    >
      <p>
        Cara Moon
      </p>
      <div>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={0}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={1}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined filled"
            id={2}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined"
            id={3}
          >
            star
          </span>
        </i>
        <i
          className="star"
        >
          <span
            className="material-symbols-outlined"
            id={4}
          >
            star
          </span>
        </i>
      </div>
    </div>
  </div>
  <div
    className="col-12"
  >
    <p>
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus convallis est sit.
    </p>
  </div>
</div>
`;
